### DISCUSSION TITLE
HEIC/HEIF support?

### DISCUSSION THREAD

**Discussion by sergey-netdev** [dlemstra/Magick.NET#670] [Help] [answered]
The documentation says "Magick.NET is linked with ImageMagick 7" and ImageMagick 7 claims to support HEIC. I can't seem to open any of HEIC/HEIF samples I have (perfectly readable by IrfanView):

> ImageMagick.MagickMissingDelegateErrorException: 'no encode delegate for this image format `HEIC' @ error/constitute.c/WriteImage/1242'

What am I missing?
p.s.
Magick.NET-Q8-AnyCPU 7.19.0.1
[Samples.zip](https://github.com/dlemstra/Magick.NET/files/4792388/Samples.zip)

**Comment by sergey-netdev** [comment]
Can also add that the following ImageMagick opens them fine:


> Version: ImageMagick 7.0.10-19 Q16 x64 2020-06-12 http://www.imagemagick.org
> Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
> License: http://www.imagemagick.org/script/license.php
> Visual C++: 180040629
> Features: Cipher DPC Modules OpenCL OpenMP(2.0)
> Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib

**Comment by dlemstra** [comment, marked as answer]
The message `'no encode delegate for this image format HEIC'` means that you cannot encode the image as a HEIC image. You can read the image as a HEIC image but you cannot write it as an HEIC image. You will need to use a different output format.

**Reply by bakertn** [reply]
I just tried to convert a JPG to HEIC and an also tried the HEIF selection.  Same error as described above, over three years later -  6/28/2023.  So there will never be any support to convert other formats to HEIF format files???

**Reply by dlemstra** [reply]
The `avif` format is supported but `heic/heif` isn't due to licensing issues.

**Reply by alexey-leonovich** [reply]
@dlemstra Is there any documentation which formats in Magick.NET are read-write and which are read-only (readonly on certain OS)? I can see only ImageMagick [formats page](https://imagemagick.org/script/formats.php). And there `.heic` is marked as "requires the [libheif](https://github.com/strukturag/libheif) delegate library" - it's unclear that actually "writing to `.heic` is [not supported](https://github.com/ImageMagick/ImageMagick/issues/1648#issuecomment-1767067199) under Windows". Thank you.

**Reply by dlemstra** [reply]
You can get the information (at runtime) with the `SupportedFormats` property of the `MagickNET` class.
